controllers/toolchaincluster/toolchaincluster_controller.go

```diff
@@ -54,29 +57,38 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	cachedCluster, ok := cluster.GetCachedToolchainCluster(toolchainCluster.Name)
 	if !ok {
 		err := fmt.Errorf("cluster %s not found in cache", toolchainCluster.Name)
-		toolchainCluster.Status.Conditions = []toolchainv1alpha1.Condition{clusterOfflineCondition()}
-		if err := r.Client.Status().Update(ctx, toolchainCluster); err != nil {
-			reqLogger.Error(err, "failed to update the status of ToolchainCluster")
+		if err := r.updateStatus(ctx, toolchainCluster, clusterOfflineCondition()); err != nil {
+			reqLogger.Error(err, "unable to update cluster status of ToolchainCluster")
```

Comment:
we can add this test for testing L61:

```go
	t.Run("error while updating a toolchain cluster on cache miss", func(t *testing.T) {
		// given
		unstable, _ := newToolchainCluster("unstable", tcNs, "http://cluster.com", toolchainv1alpha1.ToolchainClusterStatus{})
		unstable.Status.Conditions = condition.AddOrUpdateStatusConditionsWithLastUpdatedTimestamp(unstable.Status.Conditions, healthy())

		cl := test.NewFakeClient(t, unstable)
		suerr := fmt.Errorf("my test error")
		cl.MockStatusUpdate = func(ctx context.Context, obj runtimeclient.Object, opts ...runtimeclient.UpdateOption) error {
			return suerr
		}
		unstable.Status.Conditions = condition.AddOrUpdateStatusConditionsWithLastUpdatedTimestamp(unstable.Status.Conditions, clusterOfflineCondition())
		controller, req := prepareReconcile(unstable, cl, requeAfter)

		// when
		recresult, err := controller.Reconcile(context.TODO(), req)

		// then
		require.EqualError(t, err, "cluster unstable not found in cache")
		require.Equal(t, reconcile.Result{}, recresult)

		actualtoolchaincluster := &toolchainv1alpha1.ToolchainCluster{}
		err = cl.Client.Get(context.TODO(), types.NamespacedName{Name: "unstable", Namespace: tcNs}, actualtoolchaincluster)
		require.NoError(t, err)
		assertClusterStatus(t, cl, "unstable", healthy())
	})
```

Reply:
Thanks for this, i was trying to have this check in the same test as cache not found.. but i guess keep this separate would be good